openedx/core/lib/request_utils.py

```diff
@@ -80,3 +88,32 @@ def course_id_from_url(url):
         return CourseKey.from_string(course_id)
     except InvalidKeyError:
         return None
+
+
+class CookieMetricsMiddleware(MiddlewareMixin):
+    """
+    Middleware for monitoring the size and growth of all our cookies, to see if
+    we're running into browser limits.
+    """
+    def process_request(self, request, response):
+        """
+        Emit custom metrics for cookie size values for every cookie we have.
+
+        Don't log contents of cookies because that might cause a security issue.
+        We just want to see if any cookies are growing out of control.
+        """
+        if not newrelic:
+            return
+
+        cookie_names_to_size = {
+            name: len(value)
+            for name, value in request.COOKIES.items()
+        }
+        for name, size in cookie_names_to_size.items():
+            metric_name = 'cookies.{}.size'.format(name)
+            newrelic.agent.add_custom_parameter(metric_name, size)
```

Comment:
Is there a limit on the number of custom parameters?

Reply:
We discussed in person, but for record keeping purposes, I think the answer was 250 per transaction.